internal/events/persist_batch.go

```diff
@@ -298,6 +299,8 @@ func (em *eventManager) persistBatchContent(ctx context.Context, batch *fftypes.
 				log.L(ctx).Errorf("Failed to insert message entry %d in batch '%s': %s", i, batch.ID, err)
 				return false, err // a persistence failure here is considered retryable (so returned)
 			}
+			mm := matchedMsgs[i]
+			em.data.UpdateMessageCache(mm.message, mm.data)
```

Comment:
We have a postcommit hook for updating the messages via the optimized insert, but if we resort to individual upserts because the optimization failed we should update the cache after those individual upserts is my understanding.

Reply:
Great point 👍 
However, this should be a post-commit on the `UpsertMessage` call if that's ok, because we don't want the cache to be present (on the non-sending node) and visible to the aggregator until after we've successfully committed the write.

Reply:
(You'll likely have to add the `hooks` option to the call in `pkg.database.Plugin` and `internal.database`)